src/util/bip32.rs

```diff
@@ -502,13 +503,7 @@ impl ExtendedPrivKey {
             depth: 0,
             parent_fingerprint: Default::default(),
             child_number: ChildNumber::from_normal_idx(0)?,
-            private_key: PrivateKey {
-                compressed: true,
-                network: network,
-                key: secp256k1::SecretKey::from_slice(
-                    &hmac_result[..32]
-                ).map_err(Error::Ecdsa)?,
-            },
+            private_key: secp256k1::SecretKey::from_slice(&hmac_result[..32]).map_err(Error::Ecdsa)?,
```

Comment:
We have a `From<secp256k1::Error> for Error` implementation so I think you should be able to get rid of the `map_err`:

```suggestion
            private_key: secp256k1::SecretKey::from_slice(&hmac_result[..32])?,
```

Reply:
Fixed (in many other places as well)